File: pokemon_2/src/main/java/com/guivicj/pokemon_2/models/Card.java
package com.guivicj.pokemon_2.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Card {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    private String name;
    private String imageUrl;
    private String type;
    private String rarity;
}

File: pokemonClient/app/src/main/java/com/guivicj/pokemonclient/data/models/Card.kt
package com.guivicj.pokemonclient.data.models

import com.google.gson.annotations.SerializedName

data class Card(
    @SerializedName("name") val name: String,
    @SerializedName("type") val type: String,
    @SerializedName("rarity") val rarity: String,
    @SerializedName("imageUrl") val imageUrl: String
)
